Scripts/jazzy.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Generates Jazzy documentation: https://github.com/realm/jazzy
+
+jazzy \
+  --objc \
+  --author Ably \
+  --module-version 1.2.10 \
+  --umbrella-header Source/Ably.h \
+  --framework-root . \
+  --module Ably \
+  --exclude="*/ARTPushActivationStateMachine.h","*/ARTPushActivationStateMachine+Private.h" \
```

Comment:
I must admit that I am rather confused about how header files work in our project. There seem to be so many different lists of header files. To name a few:

- those specified in `Ably.podspec`
- those specified in `Package.swift`
- the directory hierarchy and header file symlinks under the `Source` directory
- `Source/Ably.modulemap` contains a list of header files
- `Source/include/Ably.modulemap` contains a list of header files
- `Source/Ably.h` contains a list of header files

And I don't fully understand what each one is used for. But I imagine that, somewhere contained amongst all those, already lies a description of all of the public header files of our SDK. It would be great if we could make use of that information in this script instead of generating _another_ list of header files.

Reply:
Jazzy docs say "In Objective-C mode, Jazzy documents all declarations found in the --umbrella-header header file and any other header files included by it". So even if you provide a "clean" list of public headers, if these headers include some private ones, they will be included in the output. Thus providing a list of exclusions is necessary. But it can be just one folder path if we tweak a project structure a bit. For example, `ARTPushActivationStateMachine` isn't considered private interface, but it shouldn't be used by users either, so it's rather `internal`. If we put these kind of classes into `internal folder`, its path, along with private headers path, can be fed to `jazzy` command.

Reply:
Well, I've made further research and now I think there is no need to split files between internal and private. Since the state machine is not used by users directly, it makes sense to put it in the private lists (and remove from `Ably.h`). See last 5 commits for details. @lawrence-forooghian
https://sdk.ably.com/builds/ably/ably-cocoa/pull/1428/jazzydoc/